socket/handlers/matching/matchingHandler/matchingStartedHandler.js

```diff
@@ -0,0 +1,143 @@
+    }
+
+    // 내 소켓에 다른 사용자 우선순위 값 넣기
+    for (const item of priorityList) {
+        if(!socket.priorityTree.contains(item.memberId)){
+            socket.priorityTree.insert(item.memberId, item.priorityValue);
+        }
+    }
+
+    // 내 소켓의 우선순위 최댓값 노드 변경
+    if (socket.priorityTree.root) {
+        socket.highestPriorityNode = socket.priorityTree.getMax(socket.priorityTree.root);
+    }
+
+    console.log('==================================================');
+    console.log(`Socket (${socket.memberId}) Priority Tree (sorted):`, JSON.stringify(socket.priorityTree.getSortedList(), null, 2));
+    console.log('Highest Priority Member:', socket.highestPriorityNode.memberId);
+    console.log('Highest Priority Value:', socket.highestPriorityNode.priorityValue);
+}
+
+/**
+ * 다른 사용자의 우선순위 트리 갱신
+ * @param {*} io 
+ * @param {*} socket 
+ * @param {*} otherPriorityList 
+ * @returns {boolean} 중복 여부
+ */
+async function updateOtherPriorityTrees(io, socket, otherPriorityList) {
+    for (const item of otherPriorityList) {
+        const otherSocket = await getSocketIdByMemberId(io, item.memberId);
+
+        if (otherSocket) {
+            if (!otherSocket.priorityTree) {
+                otherSocket.priorityTree = new PriorityTree();
+            }
+
+            // 다른 사용자 소켓에 내 우선순위 값 넣기
+            if(!otherSocket.priorityTree.contains(socket.memberId)){
+                otherSocket.priorityTree.insert(socket.memberId, item.priorityValue);
+            }
+            // 다른 사용자의 우선순위 최댓값 노드 변경
+            if (otherSocket.priorityTree.root) {
+                otherSocket.highestPriorityNode = otherSocket.priorityTree.getMax(otherSocket.priorityTree.root);
+            }
+
+            console.log('==================================================');
+            console.log(`Other Socket (${otherSocket.memberId}) Priority Tree (sorted):`, JSON.stringify(otherSocket.priorityTree.getSortedList(), null, 2));
+            console.log('Other Socket Highest Priority Member:', otherSocket.highestPriorityNode.memberId);
+            console.log('Other Socket Highest Priority Value:', otherSocket.highestPriorityNode.priorityValue);
+        }
+    }
+
+    console.log('Other Priority Trees updated based on response.');
+}
+
+/**
+ * 매칭 상대 찾기
+ * @param {*} socket 
+ * @param {*} io 
+ * @param {*} value 
+ * @returns 
+ */
+async function findMatching(socket, io, value) {
+    if (socket.highestPriorityNode !== null) {
+        // 우선순위 값 55를 넘는 모든 소켓 확인하기
+        while (socket.highestPriorityNode.priorityValue >= value) {
+            const otherSocket = await getSocketIdByMemberId(io, socket.highestPriorityNode.memberId);
+            if (otherSocket && otherSocket.highestPriorityNode.priorityValue >= value) {
+                console.log("MATCHING FOUND");
+                socket.emit("matching_found", {
+                    myMemberId: socket.memberId,
+                    otherMemberId: socket.highestPriorityNode.memberId
+                });
```

Comment:
여기서 socket event emit할 때에는 socket/common/responseFormatter로 이벤트 응답 형식을 통일해주시면 더 좋을 것 같아요!

```suggestion
  socket.emit("my-message-broadcast-success", formatResponse("my-message-broadcast-success", data));
```

이런식으로 이벤트 이름과 같이 보낼 data를 넘겨주면 됩니다!

Reply:
그리고 matching_found 이벤트는 내 소켓 + 매칭 상대방 소켓에게도 보내주어야 할 것 같아요..!